### ISSUE TITLE
Service Profile Traffic Monitor displays names which are not matching for the same Service Profile

### ISSUE THREAD

**Issue by zhasaan** [paregupt/ucs_traffic_monitor#56]
I have installed UTM version 0.3 and found out that under "Service Profile Traffic Monitor" , when you select a service profile , which is a b200M4 blade  in 5108DC= chassis in our case, it has correct information of blade# (e.g. blade-1) in "Server" column of dashboard, but then it displays a static value(blade-2) for each Service Profile in the Pictorial view of Fabric Interconnect , Chassis and Blade.
I have not seen this neither an addresses issue in UTM version 0.4 nor in closed issues list. So to my understanding it's a new issue.

Regards,
Zia-ul-Hassan

**Comment by paregupt** [comment]
I don't see this issue on my installation.
Can you please share a screenshot? Did you try the latest?

**Comment by zhasaan** [comment]
Certainly this issue is real and I see it in utm_0.3 and also after running upgrade_utm.sh, it persists. Also running the upgrade scripts breaks a few frames in the dash-

![UTM_Service_Profile_monitoring_issue](https://user-images.githubusercontent.com/15827544/115953568-99f5c980-a52f-11eb-8845-4a189cdc3afd.jpg)

**Comment by paregupt** [comment]
Thanks for sharing the screenshot. This is not the latest. The latest looks like the images in the repo. If the upgrade script didn't upgrade it correctly, please do it manually. Download the json files from the dashboard folder and import in Grafana. Make sure that UID of dashboard doesn't change.

**Comment by zhasaan** [comment]
Hi Paregupt,
Thanks for your response and my understanding was that "upgrade" script does do the sanity check first to make sure that what it needs (e.g. images in repos)  and then it executes.
Anyway I may be wrong. Now when you say manually upgrade the dashboard, is it to import all the .json files located in
https://github.com/paregupt/ucs_traffic_monitor/tree/master/grafana/dashboards
Also is there HTTP/HTTPS Proxy settings in the UTM dashboard? And finally how to verify which version of UTM is running from the dashboard.

Thanks,
Zia-ul-Hassan

**Comment by paregupt** [comment]
Looks like the upgrade script did not work in your case. The console logs may have hints why it failed. You can try running the script again.


The version is mentioned in the log files.

After the upgrade the UI will look similar to the the images on the repo. The heading text color should be blue, many more use cases, etc.

**Comment by paregupt** [comment]
Proxy setting should be done on the OS. If you used OVA, try editing the /etc/environment. Refer to the post on since2k7.com for example.

**Comment by zhasaan** [comment]
Thanks Paregupt,
I imported the new dashboards .json files into installed UTM and it has rectified the initial issue where different blades in the Chassis were coming with same number. Now my question is
1:- Is there a User Guide for this tool ?
2:- As I understood from the local Cisco Accounts team, you started this GITHUB project based on the what Customer were looking OR missing in terms of their UCS Fabric monitoring. What is the lifecycle plan for this project in long term?

What I am seeing now (even it was the case with old version of welcome.json) , that some scripts are not running correctly. Please see the attached screenshot and advise me as what and where it need to be fixed.

Many thanks,
Zia-ul-Hassan
![welcome-json](https://user-images.githubusercontent.com/15827544/116961933-f8693780-ace7-11eb-8ee9-e8601fa7c122.JPG)

**Comment by paregupt** [comment]
Great to know that the issue has been resolved.
No user-guide. UTM is self-documented. Look for 'i' icon on top left of panels. You can find detailed explanations with images. For example 
![enter image description here](https://www.since2k7.com/wp-content/uploads/2020/07/utm_0.4-10.png)

UTM is not supported by Cisco. It is my weekend project. Honestly, I know customers using it for 14+ months without missing any heartbeat. I can try my best to help you here, but no SLA.

Regarding the Welcome dashboard, it is not needed in the latest version. You can make the locations dashboard as your home dashboard. The upgrade script should have set the locations dashboard as your home dashboard but doesn't look like it worked for you.

**Comment by zhasaan** [comment]
Hi Paresh,
Thanks again for taking time respond to my query. Can I please also ask as how to change the default dashboard settings in UTM configuration?
Initial analysis of what we can monitor and see inside our UCS infrastructure using UTM in Labs seems impressive. Now next question is what expectation should we have about continuity and support(excluding SLA) of UTM if we start using in Production environment.

**Comment by paregupt** [comment]
To change the default dashboard, mark it favorite by clicking on the star icon next to the name of the dashboard on top left. Then, go to use preferences (user icon on left bottom). Choose the favorite marked dashboard from the Home Dashboard menu.

I am glad you find it impressive. 

Regarding your question, UTM has been my weekend project and till now, I have been able to spare time for it. But I can't promise support. In Cisco, a supported feature means that you can call TAC if something doesn't work. UTM doesn't have that luxury. The bottom line is that all the users of UTM are still Cisco customers and as a TME in Cisco (not in the UCS business unit), your success is my success. But UTM is not an official program. It takes my family time during weekends or late nights. With a hectic day job which is mostly spreads 1.5x and 2 young kids at home, you can imagine how much time I can spare. I hope you understand.

On the other hand, 100+ customers are using UTM in production. Some are monitoring 50+ domains and 2200+ servers. I am sure there are many more because I am not tracking anybody. I am just sharing the count that I know when someone reached out to me, just like you did. In the early days, customers like you found some issues, and they shared their environment with me to work on a fix. They also helped in verifying the fix. No major issues have been reported in the last few months. That makes me believe that the code is stable.

Using UTM in production is your decision. I have shared both the sides of the equation to help you in making a better informed decision. 

I hope this helps.

**Comment by zhasaan** [comment]
Hi Paresh,
I really appreciate your time and effort responding to my questions. I will showcase the current LABS install of UTM to my wider team and Tech Product Owner to get their viewpoint. I have added your comments to my JIRA card that I am using to carryout the work in Labs.
The local Cisco Team has also given a demo of  "Cisco Intersight" which is Cisco commercial product and  does more than just monitoring/reporting of UCS Infrastructure.
It's now up to the Team's TPO to make a decision as which path to take. We actually purchased licenses for Cisco "Performance Manager"  but never get to point of installing/configuring it. Now this product has reached EOL/EOSL.

Kind Regards,
Zia-ul-Hassan
